packages/dataviews/src/types.ts

```diff
@@ -525,37 +525,25 @@ export interface SupportedLayouts {
 	table?: Omit< ViewTable, 'type' >;
 }
 
-export interface CombinedFormField< Item > extends CombinedField {
-	render?: ComponentType< { item: Item } >;
-}
-
-export interface DataFormCombinedEditProps< Item > {
-	field: NormalizedCombinedFormField< Item >;
-	data: Item;
-	onChange: ( value: Record< string, any > ) => void;
-	hideLabelFromVision?: boolean;
-}
-
-export type NormalizedCombinedFormField< Item > = CombinedFormField< Item > & {
-	fields: NormalizedField< Item >[];
-	Edit?: ComponentType< DataFormCombinedEditProps< Item > >;
+export type FormField = {
+	id: string;
+	label?: string;
+	layout?: 'regular' | 'panel';
+	labelPosition?: 'side' | 'top';
+	children?: Array< FormField | string >;
```

Comment:
It's weird to me that "children" is part of FormField. In other words, this would make more sense to me:

```
type SimpleFormField = {
   id: string; // could be named "field" as well if we want to match "filters"
   layout: 'regular' | 'panel';
   labelPosition: 'side' | 'top';
}

type CombinedFormField = {
   id: 'string'; // actually not sure this is necessary as it's just a random string but maybe it can be useful 
   label: 'string';
   children: Array<FormField>
}

type FormField = string | SimpleFormField | CombinedFormField
```

It is possible that I'm misunderstanding combined fields, but for me, it's weird to make a field children of another field

Reply:
I see the example of "status" with "password".  Is "password" a child of "status" or we're creating a "virtual combined field" that combines both in a single panel. I feel the latter is closer to reality. 

I guess the main difference is that I'm forcing the "label" and "children" to be required for "combined fields" and I'm removing "label" for regular fields because the label is inherited from the field directly and also preventing "regular" fields to have childre.

Reply:
I might be getting into details that don't matter much at the moment, so feel free to take this comment with a grain of salt (fine to ignore for now)

Reply:
No worries of digging into the details, this part of the API is what probably needs the most fine tuning anyhow.

> It is possible that I'm misunderstanding combined fields, but for me, it's weird to make a field children of another field

True, not ideal, and I like your suggestion of separating it out again. I was running into this in the code as well, in particular with the `panel` layout as I was adding the field it-self to its children: https://github.com/WordPress/gutenberg/blob/ddf996a583b00052e1e8e0e031c616eb026c0f37/packages/dataviews/src/dataforms-layouts/panel/index.tsx#L83

The panel layout is actually what I have been struggling with the most in regards to this, as it goes a bit against the grain compared to the regular layout.
For example in the regular form, it would make sense to wrap children up in combined fields, as all is required is a label and a wrapper in essence.
The panel requires the label, but also the render method, but in the status case we display a single `value` and render multiple edit fields ( already requiring some custom logic ).

So with using the above `CombinedFormField` and `panel` layout, we may still need a reference to a `render` method. In which case I defaulted to the field referenced as part of the `id`. But maybe the first one in the `children` list would suffice?

_Sorry that may came across confusing._

> I see the example of "status" with "password". Is "password" a child of "status" or we're creating a "virtual combined field" that combines both in a single panel. I feel the latter is closer to reality.

Its the latter, a "virtual combined field". You can see my slight frustration above with how this exactly fits within the current API structure.

Reply:
> So with using the above CombinedFormField and panel layout, we may still need a reference to a render method. In which case I defaulted to the field referenced as part of the id. But maybe the first one in the children list would suffice?
> Sorry that may came across confusing.

No it actually make sense. I'm fine with using the first field to start with (I like starting small), but it does highlight that maybe at some point a custom render function receiving the children could be defined (like concatenating the values or things like that) :)

Reply:
I added the above type as part of this commit: https://github.com/WordPress/gutenberg/pull/66531/commits/21f7f9ed85d40f1d45436ad32dce75887727a47e
It works ok, at the moment it did require a few type checks in additional places. So there is room for cleanup, but that could be also be done as a follow up IMO.

Currently I have both layout fields ( regular & panel ) handle the combined fields. But I wonder if I can create a better abstraction for combined fields that would use the regular & panel under the hood. Again easier for the regular view instead of the panel view.

Also the only real difference for the panel view is that we use the first child for the `render` view. This is not a 100% clear from the code, and would require some clarification.